WcaOnRails/app/views/competitions/_index_competitions_list.html.erb

```diff
@@ -0,0 +1,56 @@
+<% if params[:display] == "list" %>
+  <% if competitions.empty? %>
+    <div class="col-md-12">
+      <div class="alert alert-warning">
+        <%= unless params[:event_ids].empty?
+          "We didn't find any competitions with those #{ pluralize(params[:event_ids].count, "event" )}! Try searching for fewer events."
+        else
+          "No competitions found."
+        end %>
+      </div>
+    </div>
+  <% elsif params[:display] == "list" %>
+    <% if params[:state] == "past" %>
+      <div class="col-md-12" id="past-comps">
+        <%= render 'index_table', competitions: competitions, title: "Past competitions from #{params[:year]}" %>
+      </div>
+    <% else %>
+      <% in_progress_competitions, upcoming_competitions = competitions.partition(&:in_progress?) %>
+      <% unless in_progress_competitions.empty? %>
+        <div class="col-md-12" id="in-progress-comps">
+          <%= render 'index_table', competitions: in_progress_competitions, title: "Competitions in progress" %>
+        </div>
+      <% end %>
+      <% unless upcoming_competitions.empty? %>
+        <div class="col-md-12" id="upcoming-comps">
+          <%= render 'index_table', competitions: upcoming_competitions, title: "Upcoming competitions" %>
+        </div>
+      <% end %>
+    <% end %>
+  <% end %>
+<% else %>
+  <script>
+    $(function() {
+      var competitions = <%= competitions.map do |c|
+        {
+          id: c.id,
+          name: c.name,
+          latitude_degrees: c.latitude_degrees,
+          longitude_degrees: c.longitude_degrees,
+          cityName: c.cityName,
+          marker_date: c.start_date.to_formatted_s(:long),
+          is_over: c.is_over?,
+          url: competition_path(c)
+        }
+      end.to_json.html_safe %>;
+
+      if(window.competitionsMap == undefined) {
+        initializeCompetitionsMap();
```

Comment:
Can we change initializeCompetitionsMap() to getCompetitionsMap(), which will create or return the competitions map dpeending on if it has been created before?